syntax = "proto3";
package mapreduce;
// 定义Empty消息类型
message Empty {}

// 定义Response消息类型
message Response {
    string message = 1;
}
// 任务状态
enum TaskState {
    TASK_IDLE = 0;   // 未分配
    TASK_IN_PROGRESS = 1;  // 执行中
    TASK_DONE = 2;      // 已完成
}
// 任务类型：Map或Reduce
enum TaskType {
    MAP_TASK = 0;
    REDUCE_TASK = 1;
    NO_TASK = 2; // 无可用任务（所有任务完成或等待中）
}

message TaskResponse{
    int32 task_id = 1;          // 任务ID（唯一）
    string filename = 2;
    int32 num_map = 3;
    int32 num_reduce = 4;       // Reduce任务总数（用于哈希分片）
    TaskType task_type = 5;
}
service MasterService{
    rpc getTask(Empty) returns (TaskResponse) {}
    rpc report_task(TaskResponse) returns (Empty) {}
}